language-support/codegen-common/src/main/scala/com/digitalasset/daml/lf/codegen/conf/Conf.scala

```diff
@@ -7,8 +7,21 @@ import java.nio.file.{Path, Paths}
 
 import ch.qos.logback.classic.Level
 import com.daml.buildinfo.BuildInfo
+import com.daml.lf.data.Ref.{PackageName, PackageVersion}
 import scopt.{OptionParser, Read}
 
+sealed trait PackageReference extends Product with Serializable
+
+object PackageReference {
+  // TODO (MK): We probably want to allow package id references here
+  // but this needs to be supported in damlc first.
+  final case class NameVersion(name: PackageName, version: PackageVersion)
```

Comment:
The indirection here might be a bit overkill. I started having both ids and name/version pairs and dropped the former once I realized that this breaks because the compiler won’t pick it up. Given that I expect we want to add it I kept the indirection but also happy to remove it if someone prefers.